lib/.eslintrc.yaml

```
@@ -7,105 +7,78 @@ rules:
   no-mixed-operators:
     - error
     - groups: [[ "&&", "||" ]]
-  no-restricted-globals:
+  no-restricted-syntax:
+    # Config copied from .eslintrc.js
+    - error
+    - selector: "CallExpression[callee.object.name='assert']:not([callee.property.name='ok']):not([callee.property.name='fail']):not([callee.property.name='ifError'])"
+      message: "Please only use simple assertions in ./lib"
+    - selector: "CallExpression[callee.name='setTimeout'][arguments.length<2]"
+      message: "setTimeout() must be invoked with at least two arguments."
+    - selector: "CallExpression[callee.name='setInterval'][arguments.length<2]"
+      message: "setInterval() must be invoked with at least 2 arguments."
+    - selector: "ThrowStatement > CallExpression[callee.name=/Error$/]"
+      message: "Use new keyword when throwing an Error."
+    # Config specific to lib
+    - selector: "NewExpression[callee.name=/Error$/]:not([callee.name=/^(AssertionError|NghttpError)$/])"
+      message: "Use an error exported by the internal/errors module."
+    - selector: "CallExpression[callee.object.name='Error'][callee.property.name='captureStackTrace']"
+      message: "Please use `require('internal/errors').hideStackFrames()` instead."
+    - selector: "AssignmentExpression:matches([left.name='prepareStackTrace'], [left.property.name='prepareStackTrace'])"
+      message: "Use 'overrideStackTrace' from 'lib/internal/errors.js' instead of 'Error.prepareStackTrace'."
+    - selector: "CallExpression[callee.name='isNaN']"
+      message: "Use NumberIsNaN() primordial instead of the global isNaN() function."
+  # Custom rules in tools/eslint-rules
+  node-core/lowercase-name-for-primitive: error
+  node-core/non-ascii-character: error
+  node-core/prefer-primordials:
     - error
     - name: Array
-      message: "Use `const { Array } = primordials;` instead of the global."
     - name: ArrayBuffer
-      message: "Use `const { ArrayBuffer } = primordials;` instead of the global."
     - name: BigInt
-      message: "Use `const { BigInt } = primordials;` instead of the global."
     - name: BigInt64Array
-      message: "Use `const { BigInt64Array } = primordials;` instead of the global."
     - name: BigUint64Array
-      message: "Use `const { BigUint64Array } = primordials;` instead of the global."
     - name: Boolean
-      message: "Use `const { Boolean } = primordials;` instead of the global."
+    - name: DataView
+    - name: Date
     - name: Error
-      message: "Use `const { Error } = primordials;` instead of the global."
+      ignore:
+        - stackTraceLimit
+        - captureStackTrace
+        - prepareStackTrace
+        - isPrototypeOf
```

Once https://github.com/nodejs/node/pull/35837 gets landed, you should be able to simplify the list:
```suggestion
        - stackTraceLimit
        - prepareStackTrace
```
Regarding `prepareStackTrace`, it's not available from primordials because it's implemented in node core rather than V8. And since `stackTraceLimit` is a property, not a method, I think users expect to be able to change its value, it wouldn't make sense for Node.js to use primordials there.

---

@aduh95 Thank you for the information. I checked your PR.

---

@Leko would be able to take that suggestion now that https://github.com/nodejs/node/pull/35837 have landed?

---

It might be a good idea to also sort this alphabetically:
```suggestion
        - prepareStackTrace
        - stackTraceLimit
```

---

Done in #36017.